### ISSUE TITLE
algoritmus zobrazování výtisků stejného dne

### ISSUE THREAD

**Issue by petrkukac** [ceskaexpedice/kramerius#1098]
Šablona zobrazení výtisků periodik v Krameriovi by měla být upravena tak, aby pokud v 1 kalendářní den vyjde více výtisků, byl při pokliku na daný datum na první pozici zobrazen „běžný“ výtisk (MODS pole genre type „normal“). 

V současnosti je zobrazován výtisk, který má nižší číslo, což je však většinou Večerní či jiné specifické vydání, nikoli běžná základní řada výtisků. Např. ukázka zde, na první pozici při pokliku na datum 11. 3. 1920 Práva lidu je zobrazen Večerník č. 58, až poté následuje základní výtisk číslo 60.  [Právo lidu 11.03.1920 | Národní digitální knihovna | Digitální knihovna Kramerius (ndk.cz)](https://ndk.cz/view/uuid:93ee9f60-2dbe-11ef-a245-0050568d319f?page=uuid:54304577-2ecf-487f-87a3-04c8146ad985)

Na dalších pořadích po běžném výtisku by postupně měly být zobrazovány výtisky s tímto genre type (údaj je editován), pokud se vyskytují:

- ranní
- odpolední
- večerní
- zvláštní
- opravené
- pořadí 1
- pořadí 2 až xy

**Comment by zabak** [comment]
Problém vidím v tom, že ten údaj v metadatech není - jsou tam jen čísla a podle těch se to řadí. V případě tohoto konkrétního titulu se večení vydání jmenuje Večerník práva lidu, ale to je specifikum tohoto konkrétního titulu. Bylo by lepší / možná by stačilo kdyby se při kliknutí na datum v kalendáři, kde je více vydání v jednom dni zobrazilo meníčko, kde by si uživatel vybral, co chce vidět.

**Comment by OndrejLehrl** [comment]
@zabak ano, klidne tak, jak pisete + pri zobrazeni v mrizce uprednostnit na prvni pozici "normal" cislo..tez pri prechodu mezi cisly vpravo v sablone pomoci navigace Předchozí/Následující číslo..ale to by se hadam automaticky takto upravilo se zmenou algoritmu razeni "normal" cisel na prvni pozici..

**Comment by honza-rychtar** [comment]
Tohle jsme v K7 přidávali pravě pro tyto účely do indexu.

Jsou k tomu vyhrazeny pole
`issue.type.code` s hodnotami
"normal", "morning", "afternoon", "evening", "corrected",  "supplement", "sequence_1", "sequence_2", "sequence_3", ...
a ` issue.type.sort` s
0, 1, 2, 3, 4, 5, 11, 12, 13, ...

Indexer umí vyčítat starší způsob zápisu
```xml
<mods:physicalDescription>
  <mods:note>večerní vydání;</mods:note>
</mods:physicalDescription>
```
i nový
```xml
<mods:genre type="evening">issue</mods:genre>
```

obojí zpracuje jako
```json
"issue.type.code": "evening",
"issue.type.sort": 3,
```

Klient pak (pouze nad K7 Krameriem) použije tato pole pro seřazení výtisků a pro zobrazení typu výtisku v bublině při hoveru na den v kalendáři při více výtisku na jeden den.

> Bylo by lepší / možná by stačilo kdyby se při kliknutí na datum v kalendáři, kde je více vydání v jednom dni zobrazilo meníčko, kde by si uživatel vybral, co chce vidět.

@zabak ten rozlišující výpis po hoveru se tenkrát zvolil jako kompromis, aby nemusel člověk klikat dvakrát.

Problém teď vidím v tom, že typ výtisku je v rámci priority řazení až za číslem. 
Předěláme to tak, že priorita řazení bude datum, typ, číslo. Pak by se to mělo chovat správně.

Každopádně je potřeba mít K7.

**Comment by zabak** [comment]
@honza-rychtar nezapomeň na zobrazení toho typu všude - v mřížkovém zobrazení by se to hodilo, protože jinak nebude jasné proč jsou ta čísla na přeskáčku. V kalendářovém zobrazení by to minimálně mělo být v alt textu, když tam nebude vybírací menu, tak ať je to aspoň vidět. A v odkazech v metadatech taky. Tam by se mohlo vylepšit to odkazování na předchozí a další tak, že pokud je v jednom dni více čísel, tak by tam byly vidět odkazy na všechna čísla z daného dne.
